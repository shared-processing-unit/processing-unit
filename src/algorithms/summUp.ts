export default (data: number[] | number[][]) => {
    const flatten = ([] as number[]).concat(...data) as number[]
    return flatten.reduce((a, b) => a + b, 0)
}
