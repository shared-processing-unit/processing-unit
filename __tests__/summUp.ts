import summUp from '../src/algorithms/summUp'

describe('summUp', () => {
    it('[1,1] should be equals 2', () => {
        expect(summUp([1, 1])).toMatchSnapshot()
    })
    it('[1,1,6,0] should be equals 8', () => {
        expect(summUp([1, 1, 6, 0])).toMatchSnapshot()
    })
    it('[[1, 1, 6, 0], [1, 2, 3, 4]] should be equals 18', () => {
        expect(summUp([[1, 1, 6, 0], [1, 2, 3, 4]])).toMatchSnapshot()
    })
    it('[[1, 1, 6, 0], [1, 2, 3, -4]] should be equals 10', () => {
        expect(summUp([[1, 1, 6, 0], [1, 2, 3, -4]])).toMatchSnapshot()
    })
})
