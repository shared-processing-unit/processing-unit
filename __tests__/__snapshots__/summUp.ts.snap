// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summUp [[1, 1, 6, 0], [1, 2, 3, -4]] should be equals 10 1`] = `10`;

exports[`summUp [[1, 1, 6, 0], [1, 2, 3, 4]] should be equals 18 1`] = `18`;

exports[`summUp [1,1,6,0] should be equals 8 1`] = `8`;

exports[`summUp [1,1] should be equals 2 1`] = `2`;
